### PR TITLE
feat(content): add support for mermaid in markdown

### LINKED ISSUE #553: Support for Mermaid diagrams & charts for content.
### Which scope/s are relevant/related to the feature request?

content

### Information

[Mermaidjs](https://github.com/mermaid-js/mermaid) is a quite famous lib to create diagrams & charts from a markdown-like format. 

I would be a great addition to add the support for it to the Analog 👍

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by JeanMeche**
This commit add the support for mermaid diagrams in markdown content. 

About mermaid : https://mermaid.js.org/
About the exisit GitHub integration : https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/

I'll provide a doc update in a followup PR if this get's merged 👍

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [ ] router
- [ ] platform
- [x] content
- [ ] nx-plugin
- [ ] trpc

Closes #553

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by brandonroberts** [comment]
Thanks @JeanMeche! These changes look good. Will need to add the mermaid peer dependency to the migration also for the next release.

https://github.com/analogjs/analog/blob/main/packages/content/migrations/migration.json

**Comment by brandonroberts** [comment]
@JeanMeche something odd is going with the latest PR preview for the analog-blog. When I visit the page with mermaid loaded, the page seems to lock up and I can't navigate around anymore.

https://deploy-preview-555--analog-blog.netlify.app/

Will you confirm and take a look?

**Comment by JeanMeche** [comment]
@brandonroberts I wrap the mermaid calls into a runOutsideAngular and it seems to have fixed the issue !

**Comment by brandonroberts** [comment]
@JeanMeche confirmed it works as expected for me also!

File: packages/content/migrations/migration.json [hunk 1]
```diff
@@ -18,6 +18,16 @@
...
+    },
+    "0.2.0-beta.26": {
+      "version": "0.2.0-beta.25",
+      "description": "Adds the new peer dependencies for mermaid",
+      "packages": {
+        "mermaidjs": {
+          "version": "^10.2.4",
+          "alwaysAddToPackageJson": true
+        }
+      }
```
Comment (brandonroberts) [inline packages/content/migrations/migration.json:30]:
```suggestion
    },
    "0.2.0-beta.26": {
      "version": "0.2.0-beta.26",
      "description": "Adds the new peer dependencies for mermaid",
      "packages": {
        "mermaid": {
          "version": "^10.2.4",
          "alwaysAddToPackageJson": true
        }
      }
```

File: packages/content/package.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "marked": "^5.0.2",
     "marked-gfm-heading-id": "^3.0.4",
     "marked-highlight": "^2.0.1",
+    "mermaidjs": "^10.2.4",
```
Comment (brandonroberts) [inline packages/content/package.json:29]:
```suggestion
    "mermaid": "^10.2.4",
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks! @allcontributors add @JeanMeche for code

**Pull request merged**